funnel/views/index.py

```diff
@@ -1,22 +1,37 @@
 # -*- coding: utf-8 -*-
 
 from flask import g, render_template, redirect, jsonify
-from datetime import datetime
-from coaster.views import jsonp, load_model
-from .. import app
+from coaster.views import jsonp, load_model, render_with
+from .. import app, funnelapp
 from ..models import Profile, ProposalSpace, Proposal
 from .space import space_data
 
 
+def index_jsonify(data):
+    return jsonify(spaces=[d for d in [dict(s.current_access()) for s in data['spaces']] if d])
+
+
 @app.route('/')
+@render_with({'text/html': 'index.html.jinja2', 'application/json': index_jsonify})
 def index():
     g.profile = None
     g.permissions = []
     spaces = ProposalSpace.fetch_sorted().filter(ProposalSpace.profile != None).all()
-    return render_template('index.html.jinja2', spaces=spaces)
+    return {'spaces': spaces}
+
+
+@funnelapp.route('/', endpoint='index')
+@render_with({'text/html': 'funnelindex.html.jinja2', 'application/json': index_jsonify})
+def funnelindex():
+    g.profile = None
+    g.permissions = []
+    spaces = ProposalSpace.fetch_sorted().filter(ProposalSpace.profile != None).all()
+    # XXX: Can we just return index() ?
+    return {'spaces': spaces}
```

Comment:
@jace can we just return index()?

Reply:
You'll get the output of `render_with`, unless you pass it `_render=False`. https://coaster.readthedocs.io/en/latest/views/decorators.html#coaster.views.decorators.render_with